### DISCUSSION TITLE
J integral in MOOSE

### DISCUSSION THREAD

**Discussion by sidharthsarmah** [idaholab/moose#26875] [Q&A Modules: Solid mechanics]
Hi @GiudGiud ,

I am currently trying to implement the J-integral in MOOSE. https://mooseframework.inl.gov/modules/tensor_mechanics/FractureIntegrals.html

Could you suggest if I read this correctly?

Example
```
[DomainIntegral]
  integrals = JIntegral
  boundary = '5000'
  crack_direction_method = CrackDirectionVector
  crack_direction_vector = '0 -1 0'
  radius_inner = 'a b c d'
  radius_outer = 'b c d e'
  output_variable = 'disp_x'
  output_q = false
  incremental = true
  symmetry_plane = 1
[]
```
1. `boundary = '5000' `: which boundary does this mean?
2. `crack_direction_vector = '0 -1 0'`: It is the direction of the crack front
3.   `radius_inner = 'a b c d'
      radius_outer = 'b c d e'`: These are the start and end radius of the contours with centre at the crack tip. Please suggest.

<img width="445" alt="image" src="https://github.com/idaholab/moose/assets/110927986/086e2147-d10d-4399-9786-14b11d54d31e">

**Comment by GiudGiud** [comment]
5000 is the ID of the nodeset forming the crack

you can check the two other items by running the example. i think you are right on both

**Reply by sidharthsarmah** [reply]
Hi @GiudGiud , thanks for the info, just a question.
I mostly make my model in ABAQUS and use the input file for the mesh.

In the examples in MOOSE, the input files are in .e format. Using Paraview I can only see the output, can you suggest how do I check the .e file to understand the commands for the mesh?

**Reply by GiudGiud** [reply]
exodus files are mostly binary so you cant really examine how they were made.
